internal/controller/pipelinerollout_controller.go

```diff
@@ -270,14 +272,32 @@ func processPipelineStatus(ctx context.Context, pipeline *kubernetes.GenericObje
 
 	numaLogger.Debugf("pipeline status: %+v", pipelineStatus)
 
+	// TODO: make string comparison instead of using numaflowv1 import (it defeats the purpose of having RawExtension).
+	// Create a function to convert/parse/check string instead of numaflowv1.Phase
+	// Do this for all 3 controllers
+
 	pipelinePhase := numaflowv1.PipelinePhase(pipelineStatus.Phase)
 	switch pipelinePhase {
 	case numaflowv1.PipelinePhaseFailed:
-		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Failed")
+		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Failed", pipelineRollout.Generation)
+	case numaflowv1.PipelinePhasePaused, numaflowv1.PipelinePhasePausing:
+		// TODO: update string(pipelinePhase) when working on strng comparison changes
+		pipelineRollout.Status.MarkPipelinePausingOrPausedWithReason(string(pipelinePhase), pipelineRollout.Generation)
+		if pipeline.Generation == pipelineStatus.ObservedGeneration {
```

Comment:
we should use `<=` whenever we compare `observedGeneration` to follow the practice from k8s

Reply:
Do we know why there could ever be a case where `generation < observedGeneration`?

Reply:
This is related to [this comment](https://github.com/numaproj/numaplane/pull/92#discussion_r1667089759)

Reply:
I think it is an extra protection that `ObservedGeneration` set to a number higher than `generation` after reconciliation.